src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowEditBase.cs

```diff
@@ -181,34 +181,32 @@ protected WhereClause GetWhereClause(bool parameterize)
                 }
                 else
                 {
-                    if (cellData.RawObject is byte[] ||
-                        col.DbColumn.DataTypeName.Equals("TEXT", StringComparison.OrdinalIgnoreCase) ||
-                        col.DbColumn.DataTypeName.Equals("NTEXT", StringComparison.OrdinalIgnoreCase))
+                    if (parameterize)
                     {
-                        // Special cases for byte[] and TEXT/NTEXT types
-                        cellDataClause = "IS NOT NULL";
-                    }
-                    else
-                    {
-                        // General case is to just use the value from the cell
-                        if (parameterize)
+                        // Add a parameter and parameterized clause component
+                        // NOTE: We include the row ID to make sure the parameter is unique if
+                        //       we execute multiple row edits at once.
+                        string paramName = $"@Param{RowId}{col.Ordinal}";
+                        if (col.DbColumn.DataTypeName.Equals("TEXT", StringComparison.OrdinalIgnoreCase) ||
+                            col.DbColumn.DataTypeName.Equals("NTEXT", StringComparison.OrdinalIgnoreCase))
                         {
-                            // Add a parameter and parameterized clause component
-                            // NOTE: We include the row ID to make sure the parameter is unique if
-                            //       we execute multiple row edits at once.
-                            string paramName = $"@Param{RowId}{col.Ordinal}";
-                            cellDataClause = $"= {paramName}";
-                            SqlParameter parameter = new SqlParameter(paramName, col.DbColumn.SqlDbType)
-                            {
-                                Value = cellData.RawObject
-                            };
-                            output.Parameters.Add(parameter);
+                            // Special cases for TEXT/NTEXT types
+                            cellDataClause = $"LIKE {paramName}";
```

Comment:
Why are you using LIKE here? Without things such as wildcards this should behave the same as just using = directly

Reply:
attempting to use = apparently does not work for strings when I tested it out

Reply:
Ah interesting - yeah the types aren't compatible https://blog.sqlauthority.com/2012/12/04/sql-server-fix-error-402-the-data-types-ntext-and-varchar-are-incompatible-in-the-equal-to-operator/

The suggestion there that I generally agree with is to use CONVERT over LIKE. But in this case I think it's fine to just use LIKE to keep it simpler.

Reply:
I'd suggest adding a comment explaining why we have to special case these though (just mention that the types aren't compatible with n/varchar so direct comparison doesn't work)